src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -697,15 +697,22 @@ case object CREATE extends OpCode(0xf0, 3, 1, _.G_create) {
 
       val codeDepositGas = state.config.calcCodeDepositCost(result.returnData)
       val gasUsed = startGas - result.gasRemaining + codeDepositGas
+      val enoughGasForDeposit = gasUsed <= availableGas
 
       if (result.error.isDefined) {
         val stack2 = stack1.push(UInt256.Zero)
         state.withStack(stack2).spendGas(startGas)
-      } else if (gasUsed > availableGas) {
+
+      } else if (!enoughGasForDeposit && state.config.exceptionalFailedCodeDeposit) {
```

Comment:
That doesn't make sense. In case of exceptional halting the contents of stack and all of the current machine state become irrelevant. 

In particular, whatever is pushed to the stack could only be used by a subsequent instruction, but there will be no subsequent instruction because we're halting with OOG.

Btw, I was looking at: https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L224 - ethereum non-classic. It's interesting how much these projects diverged.